fix: Safari rendering issues with code preview when switching tabs or scrolling

metanodreamer opened the pull request:
This PR fixes #466  in Safari browsers where the code preview would display incorrect or partially cached content when:
1. Switching between tabs (e.g., from ERC20 to Stablecoin)
2. Scrolling horizontally to view the full code
3. Changing contract settings

The issue was caused by Safari's rendering engine not properly repainting/reflowing the code preview container when content changes occurred in combination with horizontal scrolling. This resulted in showing stale or mixed content from previous states.

Fix:
- Added a force re-render mechanism that temporarily removes and re-adds the code preview element
- Applied the fix to both tab changes and contract option updates
- Set minimal timeout (0ms) to ensure the DOM has a chance to properly clear before re-rendering

The actual contract code and functionality remain unchanged - this only affects how Safari renders the content.

Testing:
1. Open the wizard in Safari (mobile or desktop)
2. Switch between tabs (especially ERC20 and Stablecoin)
3. Scroll horizontally to view full code
4. Verify that code preview always shows correct and complete content
5. Verify that contract features (copy, download, etc.) work as expected

metanodreamer commented:
I have read the CLA Document and I hereby sign the CLA

metanodreamer commented:
@ericglau, I have sent the PR. Please review the code

CoveMB commented:
Looking good thanks for tacking the issue, what do you think about wrapping the duplicated logic into a function maybe called `allowRendering` it would give more context about the usage of setTimeout 

Also have you tried using built in `tick` (to avoid microtasks)?

`
tick().then(() => {
      showCode = true;
 })
`

metanodreamer commented:
thank you @CoveMB for the suggestion. Using tick() is indeed more idiomatic and better suited for this use case. I have done the following changes:
- Introduced `allowRendering()` function
- Replaced `setTimeout` with `tick()` for better state synchronization
- Applied fix to both tab changes and contract option updates

let me know if anything else needs to be changed.

CoveMB reviewed (approved):
✨

metanodreamer commented:
hi @CoveMB, thank you for the review. I cannot merge the PR. Can you merge it for me?
